src/coreclr/jit/morph.cpp

```diff
@@ -9820,95 +9820,50 @@ GenTree* Compiler::fgMorphForRegisterFP(GenTree* tree)
 //      ignoreUsedInSIMDIntrinsic - bool. If this is set to true, then this function will ignore
 //                                  the UsedInSIMDIntrinsic check.
 //
-// return value:
+// Return Value:
 //       A GenTree* which points the simd lclvar tree belongs to. If the tree is not the simd
 //       instrinic related field, return nullptr.
 //
-
 GenTree* Compiler::getSIMDStructFromField(GenTree*     tree,
                                           CorInfoType* simdBaseJitTypeOut,
                                           unsigned*    indexOut,
                                           unsigned*    simdSizeOut,
                                           bool         ignoreUsedInSIMDIntrinsic /*false*/)
 {
-    GenTree* ret = nullptr;
-    if (tree->OperGet() == GT_FIELD)
+    if (tree->OperIs(GT_FIELD))
     {
         GenTree* objRef = tree->AsField()->GetFldObj();
-        if (objRef != nullptr)
+        if ((objRef != nullptr) && objRef->OperIs(GT_ADDR))
         {
-            GenTree* obj = nullptr;
-            if (objRef->gtOper == GT_ADDR)
-            {
-                obj = objRef->AsOp()->gtOp1;
-            }
-            else if (ignoreUsedInSIMDIntrinsic)
-            {
-                obj = objRef;
-            }
-            else
-            {
-                return nullptr;
-            }
+            GenTree* obj = objRef->AsOp()->gtOp1;
 
             if (isSIMDTypeLocal(obj))
             {
                 LclVarDsc* varDsc = lvaGetDesc(obj->AsLclVarCommon());
                 if (varDsc->lvIsUsedInSIMDIntrinsic() || ignoreUsedInSIMDIntrinsic)
                 {
-                    *simdSizeOut        = varDsc->lvExactSize;
-                    *simdBaseJitTypeOut = getBaseJitTypeOfSIMDLocal(obj);
-                    ret                 = obj;
+                    CorInfoType simdBaseJitType = varDsc->GetSimdBaseJitType();
+                    var_types   simdBaseType    = JITtype2varType(simdBaseJitType);
```

Comment:
This one is going to do:
```
BYTE -> BYTE
UBYTE -> UBYTE
SHORT -> SHORT
USHORT -> USHORT
FLOAT -> FLOAT
DOUBLE -> DOUBLE
```

but also:
```
INT, UINT -> INT
LONG, ULONG -> LONG
NATIVEINT, NATIVEUINT -> I_IMPL
```

Is that going to be too conservative for `BYTE/UBYTE` and `SHORT/USHORT`?

Reply:
Maybe more concretely, given that this is converted to an "extract", does the type here even matter?

We really just care that the field access is 1/2/4/8 bytes in size and that its at an offset of the same and so we really just care about the type of the field access so that the `Extract` node has a valid `CorInfoType` and `simdSize`

Reply:
Correct, this logic quite conservative, the "base type" of the underlying local doesn't actually matter (there is a comment below noting that). The reason it is written this way is because there is currently no scenario where we need something more advanced which would support mismatched base/tree types (we only really need to handle `TYP_FLOAT`).

Basically it is working around the fact that there is no `varType2jitType`-style function to turn the `FIELD`'s type into `CorInfoType`.